target_db2/connector.py

```diff
@@ -170,8 +170,30 @@ def _adapt_column_type(
             column_name=self.quote(column_name),
             column_type=compatible_sql_type,
         )
+        table_name_split = full_table_name.split(".", 1)
+        table_name = table_name_split[-1]
+        schema_clause = (
+            f"AND tabschema = '{table_name_split[0]}'"
+            if len(table_name_split) == 2  # noqa: PLR2004
+            else ""
+        )
+        check_reorg_stmt = sa.text(
+            f"SELECT true FROM SYSIBMADM.ADMINTABINFO WHERE REORG_PENDING "
+            f"{schema_clause} AND lower(tabname) = '{table_name}';"
+        )
```

Comment:
You could replace f-string with bind_params if you don't want to have S608 issues:
```
        schema_clause = (
            "AND tabschema = :schema_name "
            if len(table_name_split) == 2  # noqa: PLR2004
            else ""
        )
        query = "SELECT true FROM SYSIBMADM.ADMINTABINFO WHERE REORG_PENDING "
        bind_params = {"table_name": table_name}
        if schema_clause:
            bind_params.update({"schema_name": table_name_split[0]})
            query += schema_clause
        query += "AND lower(tabname) = :table_name;"
        check_reorg_stmt = sa.text(query).bindparams(**bind_params)
```
This results in statements like:

- `SELECT true FROM SYSIBMADM.ADMINTABINFO WHERE REORG_PENDING AND tabschema = 'example_schema' AND lower(tabname) = 'example_table';`
- `SELECT true FROM SYSIBMADM.ADMINTABINFO WHERE REORG_PENDING AND lower(tabname) = 'example_table';`